web/views.py

```
@@ -305,6 +310,160 @@ def index(request):
+        context = super().get_context_data(**kwargs)
+        # Check if user has already submitted this survey
+        context["already_submitted"] = Response.objects.filter(
+            user=self.request.user, question__survey=self.object
+        ).exists()
+        # Check if user is the creator of this survey
+        context["is_creator"] = self.object.author == self.request.user
+        return context
+
+
+@login_required
+def submit_survey(request, survey_id):
```

Ruff (0.8.2)

Missing return type annotation for public function `submit_survey`

(ANN201)